src/pspm_con1.m

```diff
@@ -1,176 +1,176 @@
-end;
-
-% store for output
+  return
+end
+% 2.5 store for output
 out_datatype = datatype;
 switch datatype
   case 'param'
     datatype = 'stats';
   case {'cond', 'recon'}
   otherwise
     warning('Unknown datatype');
-    return;
-end;
-
-% set load1_options
+    return
+end
+% 2.6 set load1_options
 load1_options = struct('zscored',0);
 if isfield(options, 'zscored') && options.zscored
   load1_options.zscored = 1;
-end;
-
-% work on contrasts
-% ------------------------------------------------------------------------
-for iFn =1:numel(modelfile)
-  % user output --
+end
+%% 3 work on contrasts
+for iFn = 1:numel(modelfile)
+  % 3.1 user output --
   fprintf('Loading data ... ');
-
-  % retrieve stats --
-  [sts, data, mdltype] = pspm_load1(modelfile{iFn}, datatype, '', load1_options);
-  if sts == -1
+  % 3.2 retrieve stats --
+  [lsts, data, ~] = pspm_load1(modelfile{iFn}, datatype, '', load1_options);
+  if lsts == -1
     warning('ID:invalid_input', 'Could not retrieve stats');
     return;
-  end;
-  % create con structure or retrieve existing contrasts --
+  end
+  % 3.3 create con structure or retrieve existing contrasts --
   if deletecon == 1
     con = []; conno = 0;
   else
-    [sts, con] = pspm_load1(modelfile{iFn}, 'con');
-    if sts == -1
+    [lsts, con] = pspm_load1(modelfile{iFn}, 'con');
+    if lsts == -1
       fprintf(' Creating fresh contrast structure.\n');
       con = []; conno = 0;
     else
       conno = numel(con);
-    end;
-  end;
-
-  % user output --
+    end
+  end
+  % 3.4 user output --
   fprintf('\nWriting contrasts to %s\n', modelfile{iFn});
-  % check number of contrast weights --
+  % 3.5 check number of contrast weights --
   paramno = size(data.stats, 1);
-  for c=1:numel(convec)
+  for c = 1:numel(convec)
     if numel(convec{c}) > paramno
       warning('Contrast (%d) has more weights than statistics (%d) in modelfile %s', ...
-        numel(convec{c}), paramno, modelfile{iFn}); return;
-    end;
-  end;
-  % transform into row vector and right pad with zeros --
-  conmat=zeros(numel(connames), paramno);
-  for c=1:numel(convec)
-    conmat(c,1:numel(convec{c}))=convec{c};
-  end;
-
-  % calculate contrasts --
+        numel(convec{c}), paramno, modelfile{iFn});
+      return
+    end
+  end
+  % 3.6 transform into row vector and right pad with zeros --
+  conmat = zeros(numel(connames), paramno);
+  for c = 1:numel(convec)
+    conmat(c,1:numel(convec{c})) = convec{c};
+  end
+  % 3.7 calculate contrasts
   % this automatically replicates contrasts across multiple measures if
   % data.stats has more than one column
-  conval = conmat * data.stats;
-
+  % there are issues if data.stats has NaN and the corresponding conmat
+  % also contains 0
+  idx_issue = isnan(prod(data.stats,2));
+  idx_valid_issue = transpose(conmat==0) .* idx_issue;
+  idx_invalid_issue = transpose(conmat~=0) .* idx_issue;
+  if sum(idx_issue) > 0
+    if sum(idx_valid_issue) > 0 && sum(idx_invalid_issue) == 0
```

Comment:
Here, the `sum `are run on matrices, right?
If so, then it is a bit confusing if you run an `if `on the resulting boolean **vector**

Reply:
The sum here is just used to detect if there are positive values. I have changed it to `any` as they are really Boolean vectors. In the new version, these operations are done in every line thus `idx_valid_issue` and `idx_invalid_issue` are both vectors.